core/src/Dscp/Core/Genesis.hs

```diff
@@ -59,7 +64,11 @@ distrToMap (Just addrs) (GDEqual cTotal) =
     in if sum (map (unCoin . snd) mapping) /= unCoin cTotal
        then error "distrToMap: equal summing failed"
        else GenAddressMap $ Map.fromList mapping
-distrToMap _ _ = error "distrToMap: param combination is invalid"
+distrElemToMap _ _ = error "distrToMap: param combination is invalid"
+
+distrToMap :: Maybe (NonEmpty Address) -> GenesisDistribution -> GenAddressMap
+distrToMap maddrs (GenesisDistribution distrs) =
+    fold $ fmap (distrElemToMap maddrs) distrs
```

Comment:
How exactly does it work? I would _expect_ it to sum coins if key collisions appear.